pystac_client/client.py

```diff
@@ -236,45 +299,62 @@ def from_dict(
         return result
 
     @lru_cache()
-    def get_collection(self, collection_id: str) -> Optional[Collection]:
+    def get_collection(
+        self, collection_id: str
+    ) -> Optional[Union[Collection, CollectionClient]]:
         """Get a single collection from this Catalog/API
 
         Args:
             collection_id: The Collection ID to get
 
         Returns:
-            CollectionClient: A STAC Collection
+            Union[Collection, CollectionClient]: A STAC Collection
         """
-        if self._supports_collections() and self._stac_io:
-            url = self._get_collections_href(collection_id)
+        collection: Union[Collection, CollectionClient]
+
+        if self.conforms_to(ConformanceClasses.COLLECTIONS) or self.conforms_to(
+            ConformanceClasses.FEATURES
+        ):
+            assert self._stac_io is not None
+
+            url = self._collections_href()
+            url = f"{url.rstrip('/')}/{collection_id}"
             collection = CollectionClient.from_dict(
                 self._stac_io.read_json(url),
                 root=self,
                 modifier=self.modifier,
             )
             call_modifier(self.modifier, collection)
-            return collection
         else:
-            for col in self.get_collections():
-                if col.id == collection_id:
-                    call_modifier(self.modifier, col)
-                    return col
-
-        return None
+            if self.has_conforms_to():
+                warnings.warn(
+                    DOES_NOT_CONFORM_TO("COLLECTIONS or FEATURES"),
+                    category=DoesNotConformTo,
+                    stacklevel=2,
+                )
+            warnings.warn(FALLBACK_MSG, category=FallbackToPystac, stacklevel=2)
+            for collection in super().get_collections():
+                if collection.id == collection_id:
+                    call_modifier(self.modifier, collection)
+        return collection
 
-    def get_collections(self) -> Iterator[Collection]:
+    def get_collections(self) -> Iterator[Union[Collection, CollectionClient]]:
         """Get Collections in this Catalog
 
             Gets the collections from the /collections endpoint if supported,
             otherwise fall back to Catalog behavior of following child links
 
         Return:
-            Iterator[Collection]: Iterator over Collections in Catalog/API
+            Iterator[Union[Collection, CollectionClient]]: Collections in Catalog/API
         """
         collection: Union[Collection, CollectionClient]
 
-        if self._supports_collections() and self._stac_io:
-            url = self._get_collections_href()
+        if self.conforms_to(ConformanceClasses.COLLECTIONS) or self.conforms_to(
+            ConformanceClasses.FEATURES
+        ):
```

Comment:
Since we do this twice, maybe we should keep `_supports_collections()` and just put the `conforms_to` checks in there?